amqp_events/events.py

```diff
@@ -0,0 +1,111 @@
+from functools import partial
+from typing import Callable
+
+from celery import Celery, Task
+from celery.canvas import Signature
+from celery.result import AsyncResult
+from kombu import Exchange, Queue
+
+from amqp_events import tasks, defaults
+
+
+class EventsCelery(Celery):
+    task_cls = tasks.EventHandler
+
+    def __init__(self, main, *args, **kwargs):
+        super().__init__(main, *args, **kwargs)
+        self.on_after_finalize.connect(self._generate_task_queues)
+        self.on_after_finalize.connect(self._register_retry_queues)
+
+    def event(self, name: str) -> Callable[[Callable], "Event"]:
+        def inner(func):
+            return Event(self, name, func)
+
+        return inner
+
+    def handler(self, name: str, bind: bool = False):
+        return partial(self._create_task_from_handler, name=name, bind=bind)
+
+    def _create_task_from_handler(self, fun_or_cls, *, name, bind):
+        if isinstance(fun_or_cls, type) and issubclass(fun_or_cls, Task):
+            if self.Task in fun_or_cls.__bases__:
+                bases = (fun_or_cls,)
+            else:
+                bases = (fun_or_cls, self.Task)
+            attrs = {'name': name}
+            new_name = getattr(fun_or_cls, '__name__')
+            task_class = type(new_name, bases, attrs)
+            self.register_task(task_class)
+        else:
+            self.task(fun_or_cls, name=name, bind=bind)
```

Comment:
Is possible to write one handler for several events?

Reply:
You may decorate same function or task class with different event names multiple times, like this:

```python
@handler('encoder.file.done')
@handler('encoder.file.failed`)
def record_result(*args, **kwargs):
    pass
```

Reply:
Under the hood it will be multiple registered tasks with same body. 
For convenience we can pass multiple names to `@handler` decorator.

Reply:
👍